core/include/irods/private/s3_api/common_routines.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef IRODS_S3_API_COMMON_ROUTINES_HPP
+#define IRODS_S3_API_COMMON_ROUTINES_HPP
+
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Yes, it is required for the convert_time_t_to_str() function.  Otherwise you get this error:

```
/opt/irods-externals/fmt8.1.1-0/include/fmt/chrono.h:324:24: error: 'codecvt<char32_t, char, __mbstate_t>' is deprecated [-Werror,-Wdeprecated-declarations]
```
This was the suggested workaround.

Do you think we should moved this to a header/cpp file so that the other files don't inherit this?

Reply:
I think we should narrow the scope of this to just the header(s) of interest.

If it's the format.h header, we should wrap that header with the pragmas.

Here's an example from the irods/irods repo.
https://github.com/irods/irods/blob/4d4690a3e7dc90795d597e7431b3fc33306ca227/plugins/api/src/get_file_descriptor_info.cpp#L210-L213

Reply:
done